app/controllers/mixins/ems_common_angular.rb

```diff
@@ -82,6 +99,52 @@ def create
       end
     end
 
+    def queue_authentication_check
+      ems_type = model.model_from_emstype(params[:emstype])
+
+      task_opts = {
+        :action => "Validate EMS Provider Credentials",
+        :userid => session[:userid]
+      }
+
+      queue_opts = {
+        :args => [*get_task_args(ems_type)],
+        :class_name => ems_type,
+        :method_name => "raw_connect",
+        :queue_name => "generic",
+        :role => "ems_operations",
+        :zone => params[:zone]
+      }
+
+      task_id = MiqTask.generic_action_with_callback(task_opts, queue_opts)
+      task = MiqTask.wait_for_taskid(task_id, :timeout => 30)
+
+      if task.nil?
+        error_message = "Task Error"
+      elsif MiqTask.status_error?(task.status) || MiqTask.status_timeout?(task.status)
+        error_message = task.message
+      end
+
+      [!error_message.present?, error_message]
+    end
+
+    def get_task_args(ems)
+      user, password = params[:default_userid], MiqPassword.encrypt(params[:default_password])
+      case
+      when ems == ManageIQ::Providers::Openstack::CloudManager
+        auth_url = ems.auth_url(params[:default_hostname], params[:default_api_port])
+        [user, password, auth_url]
+      when ems == ManageIQ::Providers::Amazon::CloudManager
+        [user, password, :EC2, params[:provider_region], nil, true]
+      when ems == ManageIQ::Providers::Azure::CloudManager
+        [user, password, params[:azure_tenant_id], params[:subscription], params[:proxy_uri], params[:provider_region]]
+      when ems == ManageIQ::Providers::Vmware::CloudManager
+        [params[:default_hostname], params[:default_api_port], user, password, true]
+      when ems == ManageIQ::Providers::Google::CloudManager
+          [params[:project], params[:service_account], {:service => "compute"}]
+      end
+    end
+
```

Comment:
@djberg96 there was something up with my environment when this was WIP. even through the console, comparing `ManageIQ::Providers::Openstack::CloudManager === ManageIQ::Providers::Openstack::CloudManager` was returning false. Not sure what eventually happened to fix it.